File: 1-TwoSum/1-TwoSum.java
// Last updated: 04/10/2025, 15:55:25
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int sum=0;
        int i1=0;
        int i2=0;
        for(int i=0;i<nums.length;i++){
            for(int j = i+1;j<nums.length;j++){
                int c = target-nums[i];
                if(nums[j]==c){
                    return new int[]{i,j};
                }
            }
        }
        throw new IllegalArgumentException("no match found");
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 04/10/2025, 15:55:26
class Solution {
public:
    vector<int> twoSum(vector<int>&nums,int target){
        /*
        vector<int> result;
        for(auto i=nums.begin();i!=nums.end();++i){
            auto j=find(i+1,nums.end(),target-*i);
            if(j!=nums.end()){
                result.push_back(i-nums.begin());
                result.push_back(j-nums.begin());
                break;
            }
        }
        return result;
        */
        map<int,int> a;
        for(int i=0;i<nums.size();i++){
            int num = nums[i];
            int cmpl = target-num;
            auto j = a.find(cmpl);
            if(j!=a.end()){
                return {j->second,i};
            }
            a[num] = i;
        }
        return {};
    }
};
